[KED-2504] Drop registration hooks (#1074)
quantumblacklabs/kedro: docs/source/kedro_project_setup/settings.md

# Project settings

> A Kedro project's `settings.py` file contains the runtime configuration for the project, including registration of hooks and library components. This page explains how settings work and which settings are available.

Reviewer: ```suggestion
A Kedro project's `settings.py` file contains the application configuration for the project, including registration of hooks and library components. This page explains how settings work and which settings are available.
```

> A project does not _need_ to have a settings file – in fact, by default, all code in `settings.py` is commented out. In the case that settings are not supplied, Kedro chooses sensible default values. You only need to edit `settings.py` if you wish to change to values other than the defaults.

Reviewer 2: I just tried this and actually I was wrong - you do need a settings.py file. Sorry, my bad! Maybe we should change this so that we don't require a settings.py file though 🤔 
```suggestion
By default, all code in `settings.py` is commented out. In the case that settings are not supplied, Kedro chooses sensible default values. You only need to edit `settings.py` if you wish to change to values other than the defaults.
```

## Available settings

```eval_rst
+-----------------------------------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------+
| Setting                           | Default value                                           | Use                                                                                                                                                     |
+===================================+=========================================================+=========================================================================================================================================================+

> | :code:`HOOKS`                     | :code:`tuple()`                                         | Instantiate and list your `project hooks <../extend_kedro/hooks.md>`_.                                                                                  |

Reviewer: ```suggestion
| :code:`HOOKS`                     | :code:`tuple()`                                         | Instantiate and list your project hooks <../extend_kedro/hooks.md>`_.                                                                                  |
```

Author: The ` is needed to render this piece of text as a link.
<img width="300" alt="Screenshot 2021-12-07 at 12 26 12" src="https://user-images.githubusercontent.com/49397448/145028802-be515f85-a6c2-49c8-a9a0-1cd97868e989.png">

+-----------------------------------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------+

> | :code:`DISABLE_HOOKS_FOR_PLUGINS` | :code:`tuple()`                                         | List the installed plugins for which to disable auto-registry.                                                                                          |

Reviewer: ```suggestion
| :code:`DISABLE_HOOKS_FOR_PLUGINS` | :code:`tuple()`                                         | List the installed plugins for which to disable auto-registry of hooks.                                                                                          |
```

+-----------------------------------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
| :code:`SESSION_STORE_CLASS`       | :code:`kedro.framework.session.session.BaseSessionStore`| Define where to store data from a KedroSession.                                                                                             |
+-----------------------------------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+

> | :code:`SESSION_STORE_ARGS`        | :code:`{ }`                                             | Define keyword arguments to be passed to `SESSION_STORE_CLASS` constructor.                                                                 |

Reviewer 2: I think you did this because I suggested it but actually I don't like it 😅  Sorry!!
```suggestion
| :code:`SESSION_STORE_ARGS`        | :code:`dict()`                                             | Define keyword arguments to be passed to `SESSION_STORE_CLASS` constructor.                                                                 |
```

Reviewer 2: Also does `SESSION_STORE_CLASS` render ok here or do you need to do `:code:`SESSION_STORE_CLASS`? Similarly for `CONFIG_LOADER_CLASS`.

+-----------------------------------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------+
| :code:`CONTEXT_CLASS`             | :code:`kedro.framework.context.KedroContext`            | Define custom context class.                                                                                                                            |
+-----------------------------------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------+

> | :code:`CONF_SOURCE`               | :code:`conf`                                            | Define the configuration folder.                                                                                                                        |

Reviewer 2: ```suggestion
| :code:`CONF_SOURCE`               | :code:`"conf"`                                            | Define the configuration folder.                                                                                                                        |
```

+-----------------------------------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
| :code:`CONFIG_LOADER_CLASS`       | :code:`kedro.config.ConfigLoader`                       | Define the project ConfigLoader class.                                                                                                      |
+-----------------------------------+---------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+

> | :code:`CONFIG_LOADER_ARGS`        | :code:`{ }`                                             | Define keyword arguments to be passed to `CONFIG_LOADER_CLASS` constructor. These kwargs depend on the `ConfigLoader` class implementation. |

Reviewer 2: ```suggestion
| :code:`CONFIG_LOADER_ARGS`        | :code:`dict()`                                             | Define keyword arguments to be passed to `CONFIG_LOADER_CLASS` constructor. These kwargs depend on the `ConfigLoader` class implementation. |
```

# Project settings

> A Kedro project's `settings.py` file contains the settings for the project, such as library component registration, custom hooks registration, etc. This document explains how settings work and which settings are available.

Reviewer 2: ```suggestion
A Kedro project's `settings.py` file contains the runtime configuration for the project, including registration of hooks and library components. This page explains how settings work and which settings are available.
```

> ## Default settings
> A Kedro settings file doesn’t have to define any settings if it doesn’t need to. Each setting has a sensible default value. These defaults live in the module `src/<project_name>/settings.py`.

Reviewer 2: I'd just include this in the introduction rather than its own section.
```suggestion
A project does not _need_ to have a settings file – in fact, by default, all code in `settings.py` is commented out. In the case that settings are not supplied, Kedro chooses sensible default values. You only need to edit `settings.py` if you wish to change to values other than the defaults.
```

## Available settings

#### HOOKS
Default: `tuple()`

> Instantiate and list your project hooks.

Reviewer 2: Could we link this to the docs page on hooks?

Author: Done

```python
HOOKS = (ProjectHooks(),)
```

#### DISABLE_HOOKS_FOR_PLUGINS
Default: `tuple()`

List the installed plugins for which to disable auto-registry.

```python
DISABLE_HOOKS_FOR_PLUGINS = ("kedro-viz",)
```

#### SESSION_STORE_CLASS
Default: `kedro.framework.session.session.BaseSessionStore`

Define where to store data from a KedroSession.

```python
from kedro.framework.session.store import ShelveStore

SESSION_STORE_CLASS = ShelveStore
```

#### SESSION_STORE_ARGS

> Default: { } (empty dictionary)

Reviewer 2: ```suggestion
Default: `dict()`
```

> Define keyword arguments to be passed to `SESSION_STORE_CLASS` constructor.

Reviewer 2: Do we also have a docs page to link to here?

```python
SESSION_STORE_ARGS = {"path": "./sessions"}
```

#### CONTEXT_CLASS
Default: `kedro.framework.context.KedroContext`

Define custom context class.
```python
from myproject import CustomContext

CONTEXT_CLASS = CustomContext
```

#### CONF_SOURCE
Define the configuration folder. Defaults to `conf`

```python
CONF_SOURCE = "conf"
```

#### CONFIG_LOADER_CLASS
Default: `kedro.config.ConfigLoader`

Define the project ConfigLoader class.

```python
from kedro.config import TemplatedConfigLoader

CONFIG_LOADER_CLASS = TemplatedConfigLoader
```

#### CONFIG_LOADER_ARGS

> Default: { } (empty dictionary)

Reviewer 2: ```suggestion
Default: `dict()`
```

Define keyword arguments to be passed to `CONFIG_LOADER_CLASS` constructor. These kwargs depend on the `ConfigLoader` class implementation.

```python
CONFIG_LOADER_ARGS = {
    "globals_pattern": "*globals.yml",
    "base_env": "base",
    "default_run_env": "local",
}
```

#### DATA_CATALOG_CLASS
Default: `kedro.io.DataCatalog`

Define the project `DataCatalog` class.

```python
from myproject import CustomCatalog

DATA_CATALOG_CLASS = CustomCatalog

> ```

Reviewer 2: General comment on this new page: I would keep it simple and leave out all these snippets of python. Since they're just copy and pasted from settings.py I don't think really add anything here. In my opinion the value of this page would be to summarise/give a slightly different perspective on what's available in settings.py rather than just reproducing that file in markdown. Since, unlike the actual settings.py file, it's also clickable it would be nice to directly link to other pages in the documentation here also.

I would suggest doing this as a table like this:
```
```eval_rst
+-----------------------------------+---------------+----------------------------------------------------------------+
| Setting                           | Default value | Use                                                            |
+===================================+===============+================================================================+
| :code:                            | :code:        | Instantiate and list your `project hooks <../extend_kedro/hooks.md>`_.                       |
| `HOOKS`                           | `tuple()`     |                                                                |
+-----------------------------------+---------------+----------------------------------------------------------------+
| :code:`DISABLE_HOOKS_FOR_PLUGINS` | :code:        | List the installed plugins for which to disable auto-registry. |
|                                   | `tuple()`     |                                                                |
+-----------------------------------+---------------+----------------------------------------------------------------+
|                                   |               |                                                                |
+-----------------------------------+---------------+----------------------------------------------------------------+
```

Author: I like the idea of structuring it as a table. I initially took inspiration from [Django](https://docs.djangoproject.com/en/3.2/ref/settings/), but the table will look nicer. In general the page is very much still in draft mode, but I wanted to start with something.
